src/module_wrap.cc

```diff
@@ -702,35 +714,37 @@ static MaybeLocal<Promise> ImportModuleDynamically(
   Environment* env = Environment::GetCurrent(context);
   v8::EscapableHandleScope handle_scope(iso);
 
-  if (env->context() != context) {
-    auto maybe_resolver = Promise::Resolver::New(context);
-    Local<Promise::Resolver> resolver;
-    if (maybe_resolver.ToLocal(&resolver)) {
-      // TODO(jkrems): Turn into proper error object w/ code
-      Local<Value> error = v8::Exception::Error(
-        OneByteString(iso, "import() called outside of main context"));
-      if (resolver->Reject(context, error).IsJust()) {
-        return handle_scope.Escape(resolver.As<Promise>());
-      }
-    }
-    return MaybeLocal<Promise>();
-  }
-
   Local<Function> import_callback =
     env->host_import_module_dynamically_callback();
+
+  Local<PrimitiveArray> options = referrer->GetHostDefinedOptions();
+
+  Local<Value> object;
+
+  int type = options->Get(0).As<Number>()->Int32Value();
+  if (type == ScriptType::kScript) {
+    uint32_t id = options->Get(1).As<Number>()->Uint32Value();
```

Comment:
This isn't my area at all, so can't really comment - but I'd be interested if you could explain what this ID is here?

Reply:
it comes from those counters I added to env